src/web/api/src/main/java/org/locationtech/geogig/rest/repository/UploadCommandResource.java

```diff
@@ -0,0 +1,148 @@
+            // requested output format exists, use that
+            return super.resolveFormat(options, variant);
+        }
+        // if not requested, the Variant MediaType for uploads is multipart. The response needs to
+        // be XML or JSON, so let's override it
+        return DEFAULT_OUTPUT_MEDIA_TYPE;
+    }
+
+    @Override
+    public void handlePost() {
+        // Request Entity may not be available, as in the case of a file upload.
+        // as long as the entity is not null and at least transient, call post.
+        final Optional<Representation> optional = Optional.fromNullable(getRequest().getEntity());
+        if (optional.isPresent()) {
+            final Representation entity = optional.get();
+            if (entity.isAvailable() || entity.isTransient()) {
+                post(getRequest().getEntity());
+                return;
+            }
+        }
+        // just call super
+        super.handlePost();
+    }
+
+    @Override
+    public void post(Representation entity) {
+        // invoke the runCommand method on the parent and get the representation
+        // set the representation on the response
+        getResponse().setEntity(runCommand(entity, getRequest()));
+    }
+
+    /**
+     * Consumes the data sent from the client and stores it into a temporary file to be processed.
+     * This method is just looking through the request entity for form data named
+     * {@value #UPLOAD_FILE_KEY}. If present, we will consume the data stream from the request and
+     * store it in a temporary file.
+     *
+     * @param entity POSTed entity containing binary data to be processed.
+     *
+     * @return local File representation of the data streamed form the client.
+     */
+    private File consumeFileUpload(Representation entity) {
+        File uploadedFile = null;
+        // get a File item factory
+        final DiskFileItemFactory factory = new DiskFileItemFactory();
+        // set the threshold
+        factory.setSizeThreshold(UPLOAD_THRESHOLD);
+        // build a Restlet file upload with the factory
+        final RestletFileUpload fileUploadUtil = new RestletFileUpload(factory);
+        // try to extract the uploaded file entity
+        try {
+            // build a RepresentaionContext of the request entity
+            final RepresentationContext context = new RepresentationContext(entity);
+            // get an iterator to loop through the entity for the upload data
+            final FileItemIterator iterator = fileUploadUtil.getItemIterator(context);
+            // look for the the "fileUpload" form data
+            while (iterator.hasNext()) {
+                final FileItemStream fis = iterator.next();
+                // see if this is the data we are looking for
+                if (UPLOAD_FILE_KEY.equals(fis.getFieldName())) {
+                    // found it, create a temp file
+                    uploadedFile = File.createTempFile(UUID.randomUUID().toString(), ".tmp");
```

Comment:
No need to use a UUID as key to `File.createTempFile(...)`, it adds a suffix to make it unique already. Better use a prefix that gives humans a hint of what is it about, like `"GeoGig_ImportFile"` or so